cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -608,13 +614,11 @@ func isDesiredVersionValid(csp *apis.CStorPool) bool {
 	return util.ContainsString(validVersions, version)
 }
 
-type upgradeParams struct {
-	csp    *apis.CStorPool
-	client clientset.Interface
+func upgradePath(csp *apis.CStorPool) string {
```

Comment:
liked the concept of having a function/method :+1: 
It would have been better if just VersionDetails is taken as input, so that, it can be used from multiple places

Reply:
I will take this in a separate PR,  upgradepath and validation functions to a single file.